File: snippets/java/array/move/Move.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Other/File.java to edit this template
 */
package array;

import java.util.Arrays;

/**
 *
 * @author Evans Meja
 */
public class Move {

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        int sourceArray[] = new int[] { 5, 4, 3, 2, 1};
        int destinationArray[] = new int[] { 85, 40, 13, 2, 91};
        int destinationIndex = 2;
        int sourceStart = 1;
        int count = 3;
        System.out.println(Arrays.toString(sourceArray));
        System.out.println(Arrays.toString(destinationArray));

        int[] arrayCut = Arrays.copyOfRange(sourceArray,sourceStart,sourceStart+count);
        int[] array2 = Arrays.copyOfRange(sourceArray,0,sourceStart);
        int[] array3 = Arrays.copyOfRange(sourceArray,sourceStart+count,sourceArray.length);

        System.out.println(Arrays.toString(arrayCut));
        int array2Len = array2.length;
        int array3Len = array3.length;
        int[] remining_array = new int[array2Len + array3Len];
        System.arraycopy(array2, 0, remining_array, 0, array2Len);
        System.arraycopy(array3, 0, remining_array, array2Len, array3Len);

        System.out.println(Arrays.toString(remining_array));

        int[] p1 = Arrays.copyOfRange(destinationArray,0,destinationIndex);
        int[] p2 = Arrays.copyOfRange(destinationArray,destinationIndex,destinationArray.length);


        int[] final_array = new int[destinationArray.length + arrayCut.length];

        System.arraycopy(p1, 0, final_array, 0, p1.length);
        System.arraycopy(arrayCut, 0, final_array, p1.length, arrayCut.length);
        System.arraycopy(p2, 0, final_array, (p1.length + arrayCut.length) , p2.length);

        System.out.println(Arrays.toString(final_array));



    }
}

File: snippets/c/array/move/move.c
#include<stdio.h>

int main() {
    int sourceArray[] = { 5, 4, 3, 2, 1};
    int destinationArray[] = { 85, 40, 13, 2, 91};
    int destinationIndex = 2;
    int sourceStart = 1;
    int count = 3;


    int arraySize = sizeof(sourceArray)/sizeof(sourceArray[0]);

    int en = sourceStart + ( arraySize - (sourceStart + count));

    int remainingArray[en];
    int cutArray[count];

    int a = 0;
    int b = 0;
    for(int i=0;i<arraySize;i++){
        if(i >= sourceStart && i < (sourceStart+count)){
            cutArray[a] = sourceArray[i];
            a = a + 1;
        }else{
            remainingArray[b] = sourceArray[i];
            b = b + 1;
        }
    }

    printf("array 1 %d\n");
    for(int i=0;i<a;i++){
        printf("%d\n",cutArray[i]);
    }

    printf("array 2 %d\n");
    for(int i=0;i<b;i++){
        printf("%d\n",remainingArray[i]);
    }

    int finalSize = (sizeof(destinationArray)/sizeof(destinationArray[0])) +  (sizeof(cutArray)/sizeof(cutArray[0]));

    int finalArray[finalSize];

    for(int i = 0;i<destinationIndex;i++){
        finalArray[i] = destinationArray[i];
    }

    for(int i = 0;i<count;i++){
        finalArray[i+destinationIndex] = cutArray[i];
    }

    for(int i = 0;i<=en;i++){
        finalArray[i+destinationIndex+count] = destinationArray[i+destinationIndex];
    }


    printf("final array %d\n");
    for(int i=0;i<finalSize;i++){
        printf("%d\n",finalArray[i]);
    }

   return 0;
}